coalib/results/result_actions/IgnoreResultAction.py

```diff
@@ -78,19 +89,32 @@ def apply(self, result, original_file_dict, file_diff_dict, language: str,
 
         return file_diff_dict
 
-    def get_ignore_comment(self, origin, language):
+    def get_ignore_comment(self, origin, language, original_line):
         r"""
         Returns a string of Ignore Comment, depending on the language
         Supports Single Line Comments
 
-        >>> IgnoreResultAction().get_ignore_comment("Bear", "css")
+        >>> IgnoreResultAction().get_ignore_comment("Bear", "css", "xyz\n")
         '/* Ignore Bear */\n'
 
         And Multiline Comments
 
-        >>> IgnoreResultAction().get_ignore_comment("Bear", "c")
+        >>> IgnoreResultAction().get_ignore_comment("Bear", "c", "xyz\n")
         '// Ignore Bear\n'
+
+        And Appending Comments
+
+        >>>
+        IgnoreResultAction().get_ignore_comment("New", "c", "// Ignore Bear\n")
+        'New and'
         """
+        # Searches for previous ignore comment
+        ignore_regex = re.compile(r'Ignore\s(\S+)')
+        ignore_search = ignore_regex.search(original_line)
+        if ignore_search and 'Bear' in ignore_search.group(1):
+            ignore_comment = origin + ' and'
+            return ignore_comment
```

Comment:
hm what about separating functionality a bit more:
`get_ignore_comment` (which imo should be renamed to `create_ignore_comment`, but that's a different issue :P), is responsible for creating the ignore comment. It also supports doing this for multiple bears, so `origin` --> `origins` and accepts a list (/any iterable) of origins to create the ignore comment for.

Then there's another function `read_ignore_comments` (which should actually be called `get_ignore_comments` :P), which extracts all existing ignore comments from a line. So when the action wants to add ignores, following happens:
1. `read_ignore_comments` is called, which returns all origins currently ignored (together with a `SourceRange`/`TextRange` to indicate where the comment is actually located)
2. Those origins are fed to `get_ignore_comment`, and returns the assembled new comment itself.
3. If `read_ignore_comments` did not return existing comments, the new comment is appended, otherwise the old one is replaced with the help of the positional information from `read_ignore_comments` with the new combined comment.

This is a bit more flexible and easier to maintain I believe, this would actually lay the foundation for better processing of ignore comments :)

Reply:
What do you think? :)

Reply:
Yes, you are right @Makman2, your suggestion is certainly better, but don't you think your suggested design restructures the design on which ``ignore_comments`` work whereas the functionality it would offer would be the same as I have done here. This issue is for modifying ``IgnoreResultAction`` to use conjuctions. I think a separate issue must be created which involves totally restructuring the ``IgnoreResultAction`` as per your suggestion. What do you say?

Reply:
You are right @Makman2. I have made the changes accordingly, except that ``read_ignore_comments`` is being called by ``get_ignore_comment`` as I need the comment delimiter for the regex, otherwise it may give some unexpected behaviour.